packages/fetch/src/index.ts

```diff
@@ -49,16 +49,27 @@ ${
   const normalizedParams = new URLSearchParams();
 
   Object.entries(params || {}).forEach(([key, value]) => {
+    if (value === undefined) {
+      return;
+    }
     if (value === null) {
-      normalizedParams.append(key, 'null');
-    } else if (value !== undefined) {
-      normalizedParams.append(key, value.toString());
+      return normalizedParams.append(key, "null");
+    }
+    if (Array.isArray(value)) {
+      return value.forEach((v) => {
+        if (v === null) {
+          return normalizedParams.append(key, "null");
+        }
+        return normalizedParams.append(key, v.toString());
+      });
     }
+
+    return normalizedParams.append(key, value.toString());
   });`
     : ''
 }
 
-  return \`${route}${queryParams ? '?${normalizedParams.toString()}' : ''}\`
+  return \`${route}${queryParams && normalizedParams.size ? '?${normalizedParams.toString()}' : ''}\`
```

Comment:
this dosn't work we hopeful so i fixed syntax error :)

```suggestion
  ${queryParams
    ? `return normalizedParams.size ? \`${route}${'?${normalizedParams.toString()}'}\` : \`${route}\``
    : `return \`${route}\``
  }
```

and generated bellow:

```ts
return normalizedParams.size ? `http://localhost:8000/pets?${normalizedParams.toString()}` : `http://localhost:8000/pets`
```